pkg/util/strategicpatch/fields.go

```diff
@@ -0,0 +1,469 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// NOTE: The below is taken from the Go standard library to enable us to find
+// the field of a struct that a given JSON key maps to.
+//
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package strategicpatch
```

Comment:
a) This is going to need good tests.

b) I'm not sure we want to mix with borrowed code in this way. Most/all other borrowed code is copied into Godeps or third-party. Is this file changed, or just copied?